File: app/auth/permissions.py
"""
RBAC Permission Policy
----------------------
Defines which roles are allowed to perform which actions.
Used by the RBAC enforcement layer.
"""

# Role → Allowed Permissions
ROLE_PERMISSIONS = {
    # Full administrative access
    "admin": [
        "ai.query",
        "logs.read",
        "reports.read",
        "metrics.read",
        "documents.manage",
    ],
    # Developers can interact with the AI system
    "developer": ["ai.query", "metrics.read", "documents.read"],
    # Auditors can view logs and reports but cannot interact with AI
    "auditor": ["logs.read", "reports.read", "metrics.read"],
}

File: ui/ui/lib/permissions.ts
import { rolePermissions } from "./rbac"
import { getRole } from "./auth"

/**
 * Check if the current user has a permission
 */
export function can(permission: string): boolean {
  const role = getRole()

  if (!role) return false

  return rolePermissions[role]?.includes(permission) ?? false
}